assets/src/stories-editor/helpers/index.js

```diff
@@ -1508,3 +1508,30 @@ export const getCallToActionBlock = ( pageClientId ) => {
 export const getUniqueId = () => {
 	return uuid().replace( /^\d/, 'a' );
 };
+
+/**
+ * Returns an image of the first frame of a given video.
+ *
+ * @todo Perhaps allow specifying wanted image type.
+ *
+ * @param {string} src Video src URL.
+ * @return {Promise<string>} The extracted image in base64-encoded format.
+ */
+export const getFirstFrameOfVideo = async ( src ) => {
+	const video = document.createElement( 'video' );
+	video.setAttribute( 'muted', 'muted' );
+
+	return new Promise( ( resolve ) => {
+		video.src = src;
+		video.addEventListener( 'loadeddata', () => {
+			const canvas = document.createElement( 'canvas' );
+			canvas.width = video.videoWidth;
+			canvas.height = video.videoHeight;
+
+			const ctx = canvas.getContext( '2d' );
+			ctx.drawImage( video, 0, 0, canvas.width, canvas.height );
+
+			canvas.toBlob( resolve );
```

Comment:
^ Does this work for any URL-s, external included?

Reply:
There can be CORS issues with external URLs: https://developer.mozilla.org/en-US/docs/Web/HTML/CORS_enabled_image#Security_and_tainted_canvases

Reply:
Set the `crossOrigin` attribute now, which seems to resolve any issues with this.